src/detect-engine-proto.c

```diff
@@ -143,6 +143,29 @@ int DetectProtoContainsProto(const DetectProto *dp, int proto)
     return 0;
 }
 
+/* List supported detect protocol keywords */
+
+void ListSupportedDetectProtocols(void)
+{
+    SCEnter();
+    printf("=========Supported Detect Protocols=========\n");
+	printf("- tcp\n");
+	printf("- tcp-pkt\n");
+	printf("- tcp-stream\n");
+	printf("- udp\n");
+	printf("- icmpv4\n");
+	printf("- icmpv6\n");
+	printf("- icmp\n");
+	printf("- sctp\n");
+	printf("- ipv4\n");
+	printf("- ip4\n");
+	printf("- ipv6\n");
+	printf("- ip6\n");
+	printf("- ip\n");
+	printf("- pkthdr\n");
+    SCReturn;
```

Comment:
Should this also list the valid app protos?

Reply:
within this list but with another headline?